evm/src/fuzz/strategies/state.rs

```diff
@@ -53,20 +53,35 @@ This is a bug, please open an issue: https://github.com/foundry-rs/foundry/issue
 }
 
 /// Builds the initial [EvmFuzzState] from a database.
-pub fn build_initial_state<DB: DatabaseRef>(db: &CacheDB<DB>) -> EvmFuzzState {
+pub fn build_initial_state<DB: DatabaseRef>(
+    test_address: Address,
+    db: &CacheDB<DB>,
+) -> EvmFuzzState {
     let mut state: BTreeSet<[u8; 32]> = BTreeSet::new();
     for (address, account) in db.accounts.iter() {
+        // We don't want to collect data from the test contract.
+        if *address == test_address {
```

Comment:
hmmm i actually disagree here. A lot of times the test contract will be an owner of a protocol's contract. And people throw relevant state in the test contract as well (esp w/ invariant tests, things like target contracts, etc)

Reply:
ultimately these should probably be flags exposed in the config—there are cases where collecting data from the test contract will flood your dict, and other times where it may be valuable

Reply:
imo very unlikely it will flood your dict - we collect push bytes, storage and basic account info. a lot of those values will either be duplicate of subcontracts or be relevant to the subcontracts (e.g. their addresses, assertion values etc.). if values are duplicate they don't expand the dictionary since it's a set

Reply:
@joshieDo if you really mean `CALLER` instead of `test contract` I agree, we dont wan't the caller. But we do want test contract state + address in the dictionary

Reply:
hmm ok so e.g. stack/mem from a large setUp method or test contract helper methods wouldn't be collected? if so then I agree it's not likely to flood and should be ok

Reply:
stack/mem is in a separate file, this file only collects push bytes + storage

Reply:
`stack/mem` would be collected for invariants (through the inspector) but not normal fuzzing (there's no inspector collector).

Hmm, I see your overall point. I'll revert it. ( i really meant the test contract).